copy/copy.go

```diff
@@ -695,9 +695,11 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, ba
 			// This is safe because we have just computed diffIDResult.Digest ourselves, and in the process
 			// we have read all of the input blob, so srcInfo.Digest must have been validated by digestingReader.
 			ic.c.blobInfoCache.RecordDigestUncompressedPair(srcInfo.Digest, diffIDResult.digest)
+			bar.SetTotal(srcInfo.Size, true)
 			return blobInfo, diffIDResult.digest, nil
 		}
 	} else {
+		bar.SetTotal(srcInfo.Size, true)
```

Comment:
(Non-blocking: It would be a tiny bit more readable to consolidate the success return paths again, so that this happens only in one place.)

Reply:
I did that and removed the XXX indicator. Thanks!

Reply:
Almost… there are still two `SetTotal` calls.

Reply:
:joy_cat: omg, thanks for bearing with me. Updated.